ui/job-view/details/PinBoard.jsx

```diff
@@ -216,7 +247,49 @@ export default class PinBoard extends React.Component {
 
   async cancelAllPinnedJobs() {
     if (window.confirm('This will cancel all the selected jobs. Are you sure?')) {
-      await JobModel.cancel(this.$rootScope.repoName, Object.keys(this.props.pinnedJobs));
+      const jobIds = Object.keys(this.props.pinnedJobs);
+
+      this.$timeout(() => this.thNotify.send(
+        'Request sent to cancel all pinned jobs via actions.json',
+        'success'),
+      );
+
+      try {
+        jobIds.forEach(async (id) => {
```

Comment:
This code and that in ``ActionBar.jsx`` seem super similar.  I wonder about your thoughts of placing this function (and perhaps ``retrigger``) in the ``JobModel`` so it can be reused.  Or possibly in ``helpers/taskcluster``.  So the function would take the ``jobIds`` and I suppose it would need the ``ThResultSetStore`` too (that service will be replaced soon, but I can handle that migration at the time).

Reply:
I initially tried to do that but every time I try to import JobModel, the site would just crash for some reason.

For example, adding `import JobModel from '../models/job` in `helpers/taskcluster.js` throws:

```bash
Uncaught TypeError: __webpack_require__.i(...) is not a function
    at Object.<anonymous> (job.js?c11b:4)
    at __webpack_require__ (bootstrap 4256e1fd06053122d10f:693)
    at fn (bootstrap 4256e1fd06053122d10f:114)
    at Object.invariant (font-awesome.css?0dc0:8)
    at __webpack_require__ (bootstrap 4256e1fd06053122d10f:693)
    at fn (bootstrap 4256e1fd06053122d10f:114)
    at Object.<anonymous> (index.bundle.js:9)
    at __webpack_require__ (bootstrap 4256e1fd06053122d10f:693)
    at fn (bootstrap 4256e1fd06053122d10f:114)
    at Object.<anonymous> (index.bundle.js:47532)
(anonymous) @ job.js?c11b:4
__webpack_require__ @ bootstrap 4256e1fd06053122d10f:693
fn @ bootstrap 4256e1fd06053122d10f:114
invariant @ font-awesome.css?0dc0:8
__webpack_require__ @ bootstrap 4256e1fd06053122d10f:693
fn @ bootstrap 4256e1fd06053122d10f:114
(anonymous) @ index.bundle.js:9
__webpack_require__ @ bootstrap 4256e1fd06053122d10f:693
fn @ bootstrap 4256e1fd06053122d10f:114
(anonymous) @ index.bundle.js:47532
__webpack_require__ @ bootstrap 4256e1fd06053122d10f:693
fn @ bootstrap 4256e1fd06053122d10f:114
(anonymous) @ emitter.js:2
__webpack_require__ @ bootstrap 4256e1fd06053122d10f:693
fn @ bootstrap 4256e1fd06053122d10f:114
(anonymous) @ log-apply-result.js:30
__webpack_require__ @ bootstrap 4256e1fd06053122d10f:693
webpackJsonpCallback @ bootstrap 4256e1fd06053122d10f:25
(anonymous) @ index.bundle.js:1
angular.js:15535 Error: [$controller:ctrlreg] The controller with the name 'MainCtrl' is not registered.
https://errors.angularjs.org/1.7.3/$controller/ctrlreg?p0=MainCtrl
    at angular.js:138
    at $controller (angular.js:11645)
    at setupControllers (angular.js:10676)
    at nodeLinkFn (angular.js:10461)
    at compositeLinkFn (angular.js:9800)
    at compositeLinkFn (angular.js:9803)
    at publicLinkFn (angular.js:9665)
    at angular.js:1946
    at Scope.$eval (angular.js:19355)
    at Scope.$apply (angular.js:19454)
```

Reply:
I'll try to rebase and see if that makes any difference.

Reply:
No success.

Reply:
I can't say for SURE what is causing that without seeing the code, but you'd need to pass ``this.thNotify`` and ``this.$rootScope.repoName`` in as params to the function.  These are angular services.  So I suspect that's the problem.  Would you give that a try?

Reply:
Oh, sorry.  My bad.  OK, I can repro locally.  Looking into it.

Reply:
I'm not able to find a good reason or workaround for importing the ``JobModel`` in the ``taskcluster`` helper.  But it seems like it might be better to have the function in ``JobModel`` and have it import the ``taskcluster`` helper from there.

We are in this odd state of being MOSTLY React, but still the root of the app is in Angular, and it seems like something is having a problem in Webpack there.  But the import I stated doesn't throw that error, and it seems like that might be the right place to put these functions anyway.  

What do you think?

Reply:
That works! I'll go ahead and remove the review flag and put it back when the patch is done. Thanks for looking into it :)